### DISCUSSION TITLE
Announcement of the Deprecation of PHP language versions less than 7.2.5 in the Amazon SNS Message Validator for PHP

### DISCUSSION THREAD

**Discussion by yenfryherrerafeliz** [aws/aws-php-sns-message-validator#67] [Announcements]
On December 15, 2023, in order to continue supporting our customers with tools that are secure and maintainable, AWS will publish a minor version bump of the [Amazon SNS Message Validator for PHP](https://github.com/aws/aws-php-sns-message-validator). These new versions will require a PHP 7.2.5+ language version, formally ending our support for versions 7.2.4 and below. We've recently ended support for versions 7.2.4 and below in the [AWS SDK for PHP](https://github.com/aws/aws-sdk-php). The reason for this is that the PHP Group, stated that, as of November 30th, 2020, no new bug reports, fixes, or changes will be made to PHP 7.2.

**I use a PHP language level of 7.2.5 or higher, what should I do?**
No actions are required.
**I use a PHP language level below 7.2.5, what should I do?**
For users below 7.2.5, we recommend updating your PHP language version to a language version supported by the Amazon SNS Message Validator, if possible.

In the event that you’re unable to migrate to PHP 7.2.5 or higher, previous versions of the Amazon SNS Message Validator will continue to operate after the deprecation date. Depending on your installation method, you will either stop receiving updates after the deprecation date, or will need to download a release that is compatible with your PHP version.
Conclusion:
If you’re currently using a version of PHP below 7.2.5 as your runtime for Amazon SNS Message Validator, we encourage you to upgrade to PHP 7.2.5 or higher. Please share your questions, comments, and issues with us on [GitHub](https://github.com/aws/aws-php-sns-message-validator/issues).

**Comment by mgkimsal** [comment]
Curious why a new version isn't simply released when it's ready.  People who need an earlier version can always still go get the previous version, no?

**Comment by tinusg** [comment]
I appreciate the update, but I think there are more people running into the fact that you still require the psr/http-message:1.0 package than people who don't know how to install an old version of your package. And now we have to wait 2 weeks before we can update, while a fix is seemingly available.
